packages/shared/src/components/fields/PasswordField.tsx

```diff
@@ -48,16 +53,22 @@ export function PasswordField({
       valueChanged={(value) =>
         setPasswordStrengthLevel(passwordStrength(value).id)
       }
-      hint={hint}
+      hint={showStrength ? hint : props.hint}
       validityChanged={setIsValid}
       valid={isValid}
       hintClassName={
-        hasUserAction && passwordStrengthStates[passwordStrengthLevel].className
+        showStrength &&
```

Comment:
Seeing you use this multiple times you can actually abstract it to it's own variable in this file.
Will clean up the render bit.